// payload/dist/admin/components/elements/ArrayAction/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "ArrayAction", {
    enumerable: true,
    get: function() {
        return ArrayAction;
    }
});
const _react = /*#__PURE__*/ _interop_require_default(require("react"));
const _reacti18next = require("react-i18next");
const _Chevron = /*#__PURE__*/ _interop_require_default(require("../../icons/Chevron"));
const _Copy = /*#__PURE__*/ _interop_require_default(require("../../icons/Copy"));
const _More = /*#__PURE__*/ _interop_require_default(require("../../icons/More"));
const _Plus = /*#__PURE__*/ _interop_require_default(require("../../icons/Plus"));
const _X = /*#__PURE__*/ _interop_require_default(require("../../icons/X"));
const _Popup = /*#__PURE__*/ _interop_require_default(require("../Popup"));
const _PopupButtonList = /*#__PURE__*/ _interop_require_wildcard(require("../Popup/PopupButtonList"));
require("./index.scss");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
const baseClass = 'array-actions';
const ArrayAction = ({ addRow, duplicateRow, hasMaxRows, index, isSortable, moveRow, removeRow, rowCount })=>{
    const { t } = (0, _reacti18next.useTranslation)('general');
    return /*#__PURE__*/ _react.default.createElement(_Popup.default, {
        button: /*#__PURE__*/ _react.default.createElement(_More.default, null),
        buttonClassName: `${baseClass}__button`,
        className: baseClass,
        horizontalAlign: "center",
        render: ({ close })=>{
            return /*#__PURE__*/ _react.default.createElement(_PopupButtonList.ButtonGroup, {
                buttonSize: "small"
            }, isSortable && index !== 0 && /*#__PURE__*/ _react.default.createElement(_PopupButtonList.Button, {
                className: `${baseClass}__action ${baseClass}__move-up`,
                onClick: ()=>{
                    moveRow(index, index - 1);
                    close();
                }
            }, /*#__PURE__*/ _react.default.createElement("div", {
                className: `${baseClass}__action-chevron`
            }, /*#__PURE__*/ _react.default.createElement(_Chevron.default, {
                direction: "up"
            })), t('moveUp')), isSortable && index < rowCount - 1 && /*#__PURE__*/ _react.default.createElement(_PopupButtonList.Button, {
                className: `${baseClass}__action`,
                onClick: ()=>{
                    moveRow(index, index + 1);
                    close();
                }
            }, /*#__PURE__*/ _react.default.createElement("div", {
                className: `${baseClass}__action-chevron`
            }, /*#__PURE__*/ _react.default.createElement(_Chevron.default, null)), t('moveDown')), !hasMaxRows && /*#__PURE__*/ _react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/ _react.default.createElement(_PopupButtonList.Button, {
                className: `${baseClass}__action ${baseClass}__add`,
                onClick: ()=>{
                    addRow(index + 1);
                    close();
                }
            }, /*#__PURE__*/ _react.default.createElement(_Plus.default, null), t('addBelow')), /*#__PURE__*/ _react.default.createElement(_PopupButtonList.Button, {
                className: `${baseClass}__action ${baseClass}__duplicate`,
                onClick: ()=>{
                    duplicateRow(index);
                    close();
                }
            }, /*#__PURE__*/ _react.default.createElement(_Copy.default, null), t('duplicate'))), /*#__PURE__*/ _react.default.createElement(_PopupButtonList.Button, {
                className: `${baseClass}__action ${baseClass}__remove`,
                onClick: ()=>{
                    removeRow(index);
                    close();
                }
            }, /*#__PURE__*/ _react.default.createElement(_X.default, null), t('remove')));
        },
        size: "medium"
    });
};
