// payload/dist/uploads/cropImage.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    default: function() {
        return cropImage;
    },
    percentToPixel: function() {
        return percentToPixel;
    }
});
const _sharp = /*#__PURE__*/ _interop_require_default(require("sharp"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const percentToPixel = (value, dimension)=>{
    if (!value) return 0;
    return Math.floor(parseFloat(value) / 100 * dimension);
};
async function cropImage({ cropData, dimensions, file }) {
    try {
        const { height, width, x, y } = cropData;
        const formattedCropData = {
            height: percentToPixel(height, dimensions.height),
            left: percentToPixel(x, dimensions.width),
            top: percentToPixel(y, dimensions.height),
            width: percentToPixel(width, dimensions.width)
        };
        const cropped = (0, _sharp.default)(file.tempFilePath || file.data).extract(formattedCropData);
        return await cropped.toBuffer({
            resolveWithObject: true
        });
    } catch (error) {
        console.error(`Error cropping image:`, error);
        throw error;
    }
}
