// payload/dist/admin/components/forms/field-types/RadioGroup/RadioInput/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return _default;
    }
});
const _react = /*#__PURE__*/ _interop_require_default(require("react"));
const _reacti18next = require("react-i18next");
const _getTranslation = require("../../../../../../utilities/getTranslation");
const _EditDepth = require("../../../../utilities/EditDepth");
require("./index.scss");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const baseClass = 'radio-input';
const RadioInput = (props)=>{
    const { isSelected, onChange, option, path, readOnly } = props;
    const { i18n } = (0, _reacti18next.useTranslation)();
    const editDepth = (0, _EditDepth.useEditDepth)();
    const classes = [
        baseClass,
        isSelected && `${baseClass}--is-selected`
    ].filter(Boolean).join(' ');
    const id = `field-${path}-${option.value}${editDepth > 1 ? `-${editDepth}` : ''}`;
    return /*#__PURE__*/ _react.default.createElement("label", {
        htmlFor: id
    }, /*#__PURE__*/ _react.default.createElement("div", {
        className: classes
    }, /*#__PURE__*/ _react.default.createElement("input", {
        checked: isSelected,
        disabled: readOnly,
        id: id,
        onChange: ()=>typeof onChange === 'function' ? onChange(option.value) : null,
        type: "radio"
    }), /*#__PURE__*/ _react.default.createElement("span", {
        className: `${baseClass}__styled-radio`
    }), /*#__PURE__*/ _react.default.createElement("span", {
        className: `${baseClass}__label`
    }, (0, _getTranslation.getTranslation)(option.label, i18n))));
};
const _default = RadioInput;
